.github/workflows/canary-integration-test.yml

```diff
@@ -1468,3 +1468,44 @@ jobs:
         uses: ./.github/workflows/collect-logs
         with:
           name: csi-hostnetwork-disabled
+
+  csi-addons-testing:
+      runs-on: ubuntu-20.04
+      if: "!contains(github.event.pull_request.label.*.name, 'skip-ci')"
+      steps:
+        - name: checkout
+          uses: actions/checkout@v4
+          with:
+            fetch-depth: 0
+
+        - name: consider debugging
+          uses: ./.github/workflows/tmate_debug
+          with:
+            use-tmate: ${{ secrets.USE_TMATE }}
+
+        - name: setup cluster resources
+          uses: ./.github/workflows/canary-test-config
+
+        - name: use local disk
+          run: tests/scripts/github-action-helper.sh use_local_disk
+
+        - name: create cluster prerequisites
+          run:  tests/scripts/github-action-helper.sh create_cluster_prerequisites
+
+        - name: deploy cluster
+          run: tests/scripts/github-action-helper.sh deploy_cluster
+
+        - name: setup csi-addons
+          run:  tests/scripts/csiaddons.sh setup_csiaddons
+
+        - name: wait for prepare pod
+          run: tests/scripts/github-action-helper.sh wait_for_prepare_pod
+
+        - name: wait for ceph to be ready
+          run: tests/scripts/github-action-helper.sh wait_for_ceph_to_be_ready all 1
+
+        - name: verify csiaddonsnodes crd created
+          run:  tests/scripts/csiaddons.sh verify_crd_created
+
+        - name: verify csi-addons container running
+          run: tests/scripts/csiaddons.sh verify_container_is_running
```

Comment:
also, please add the log collection for test similar to other test in the file
https://github.com/rook/rook/blob/bb4528d460db17909b3dec0b2f31374913345cd5/.github/workflows/canary-integration-test.yml#L318-L322